express-notes-api/server.ts

```diff
@@ -0,0 +1,111 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+app.use(express.json());
+
+type Data = {
+  nextId: number;
+  notes: Record<string, string>;
+};
+
+let gradesArray: string[] = [];
+
+app.get('/api/notes', async (req, res) => {
+  const data = await read();
+  for (const key in data.notes) {
+    if (!data.notes[key]) {
+      gradesArray = [];
+      break;
+    } else {
+      gradesArray.push(data.notes[key]);
+    }
+  }
+  res.json(gradesArray);
+  gradesArray = [];
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  const data = await read();
+  const id = +req.params.id;
+  const requestedNote = data.notes[id];
+  if (!Number.isInteger(id)) {
+    res.status(400).json({ error: 'Id must be a positive integer' });
+  } else if (!requestedNote) {
+    res.status(404).json({ error: `Id ${id} was not found` });
+  } else {
+    res.json(requestedNote);
+  }
+});
+
+const includesAll = (arr: string[], values: string[]): boolean => {
+  return values.every((v) => arr.includes(v));
+};
+
+app.post('/api/notes', async (req, res) => {
+  const data = await read();
+  const newObj = req.body;
+  if (!includesAll(Object.keys(newObj), ['content'])) {
+    res.status(400).json({ error: 'Content is the only supported input' });
+  } else if (includesAll(Object.keys(newObj), ['content'])) {
```

Comment:
You already checked `includesAll` in the `if` statement, so you don't need to check it again here in the `else`.